griptape/tasks/api_request_subtask.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import annotations
+import json
+import re
+from typing import Optional, TYPE_CHECKING
+import schema
+from attr import define, field
+from jsonschema.exceptions import ValidationError
+from jsonschema.validators import validate
+from schema import Schema, Literal
+from griptape.artifacts import ErrorArtifact, TextArtifact
+from griptape.utils import remove_null_values_in_dict_recursively
+from griptape.mixins import ActivityMixin, ApiRequestSubtaskOriginMixin
+from griptape.tasks import PromptTask, BaseTask
+from griptape.artifacts import BaseArtifact
+from griptape.events import (
+    StartApiRequestSubtaskEvent,
+    FinishApiRequestSubtaskEvent,
+)
+
+if TYPE_CHECKING:
+    from griptape.memory import ToolMemory
+    from griptape.tools import BaseTool
+
+
+@define
+class ApiRequestSubtask(PromptTask):
```

Comment:
Just making sure I understand the future of this. Today, this is tightly coupled with Tool execution, and might benefit from a name of something like `ToolExecutionSubtask`. But this will grow over time to include other types of API requests, right?